js/src/logger.ts

```diff
@@ -1561,16 +1792,24 @@ class BackgroundLogger {
       return;
     }
     try {
-      const allItems = await this.unwrapLazyValues(wrappedItems);
+      const [allItems, allAttachments] =
+        await this.unwrapLazyValues(wrappedItems);
       const dataStr = constructLogs3Data(
         allItems.map((x) => JSON.stringify(x)),
       );
+      const attachmentStr = JSON.stringify(
+        allAttachments.map((a) => a.debugInfo()),
+      );
+
+      const promises: Promise<void>[] = [];
       for (const payloadDir of publishPayloadsDir) {
-        await BackgroundLogger.writePayloadToDir({
-          payloadDir,
-          payload: dataStr,
-        });
+        const write = (payload: string) =>
+          BackgroundLogger.writePayloadToDir({ payloadDir, payload });
+        promises.push(write(dataStr));
+        promises.push(write(attachmentStr));
```

Comment:
I suspect this will end up with a lot of empty `[]` entries in the debug directory. Could we either bundle the `dataStr` and `attachmentStr` into one object with keys `data` and `attachments`, or omit the attachments when the list is empty?

Reply:
> one object with keys `data` and `attachments`

Done